contracts/src/utils/TokenScalingUtils.sol

```diff
@@ -5,6 +5,8 @@
 pragma solidity 0.8.18;
 
 library TokenScalingUtils {
+    uint256 public constant MAX_TOKEN_DECIMALS = 30;
```

Comment:
@michaelkaplan13 you mentioned we should use `18` as the max, in [this message](https://github.com/ava-labs/teleporter-token-bridge/pull/142#issuecomment-2121267890), but this was the number needed  to make the existing tests pass. If we're not comfortable with this crazy `30` 😅 then I can put the work into adjusting the test logic.

Reply:
I think we should go with 18 and update the tests as needed for these changes because the existing tests didn't have any notion of specifying an explicit decimals value, so seems like that it could be an incorrect parameter value that's being used.

Reply:
I suspect the failing tests were likely because 18+3 is being used as the token decimals value in a few places, including [here](https://github.com/ava-labs/teleporter-token-bridge/pull/142/files#diff-07b24b3478a7e372a4fb5ef00ec2dc32cc1f8a0369963f00b595399f932ba513R61).

Lets keep this at 18 and fix the tests. I think we should remove `_DEFAULT_DECIMAL_SHIFT` and instead define `_DEFAULT_DESTINATION_DECIMALS` to a different value than the source's decimals default.